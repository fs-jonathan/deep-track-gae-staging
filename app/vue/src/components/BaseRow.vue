<template>
  <div class="bg-white hover:bg-grey-light m-3 max-w-sm shadow-md rounded-md overflow-hidden" v-on:click="showRecord(result.id)">
    <div class="text-left p-1 sm:text-left sm:flex-grow">
      <span class="text-sm leading-tight">{{ result.title }}</span>
      <span class="text-xs leading-tight text-grey-dark sm:inline">{{ result.subtitle }}</span>
    </div>
    <div class="p-1">
      <strong class="text-xl leading-tight">¥{{ result.cost }}</strong>
    </div>
    <div class="text-left sm:text-left sm:flex-grow p-1" v-if="result.compare > 0">
      <img src="../assets/icon_arrow_right.png" class="w-2"/>
      <span class="text-xs text-grey-dark m-1">¥{{ result.compare }}({{ result.rate }}%)</span>
    </div>
  </div>
</template>

<script>
export default {
  name: 'BaseRow',
  props: ['result'],
  methods: {
    showRecord: function(id) {
      this.$router.push({ name: 'ShowRecord', params: { index: id } });
    }
  }
}
</script>
