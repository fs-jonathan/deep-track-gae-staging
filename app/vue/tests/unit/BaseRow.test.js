import { shallowMount } from '@vue/test-utils'
import BaseRow from '@/components/BaseRow.vue'

describe('HelloWorld.vue', () => {
  it('renders correctly', () => {
    const msg = 'new message'
    const wrapper = shallowMount(BaseRow, {
      propsData: { result: {
        title: msg,
        cost: 0,
        compare: 1,
        rate: 1,
        id: 2
      } }
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
