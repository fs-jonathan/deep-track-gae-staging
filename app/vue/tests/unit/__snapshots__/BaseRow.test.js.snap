// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue renders correctly 1`] = `
<div
  class="bg-white hover:bg-grey-light m-3 max-w-sm shadow-md rounded-md overflow-hidden"
>
  <div
    class="text-left p-1 sm:text-left sm:flex-grow"
  >
    <span
      class="text-sm leading-tight"
    >
      new message
    </span>
     
    <span
      class="text-xs leading-tight text-grey-dark sm:inline"
    >
      
    </span>
  </div>
   
  <div
    class="p-1"
  >
    <strong
      class="text-xl leading-tight"
    >
      ¥0
    </strong>
  </div>
   
  <div
    class="text-left sm:text-left sm:flex-grow p-1"
  >
    <img
      class="w-2"
      src="../assets/icon_arrow_right.png"
    />
     
    <span
      class="text-xs text-grey-dark m-1"
    >
      ¥1(1%)
    </span>
  </div>
</div>
`;
